synfig-studio/src/gui/app.cpp

```diff
@@ -844,14 +845,115 @@ class Preferences : public synfigapp::Settings
 
 static ::Preferences _preferences;
 
+void
+init_builder()
+{
+	
+	Glib::RefPtr<Gio::SimpleActionGroup> simple_menus_action_group = Gio::SimpleActionGroup::create();
+	Glib::RefPtr<Gio::SimpleActionGroup> simple_actions_action_group = Gio::SimpleActionGroup::create();
+
+	simple_menus_action_group->add_action("menu-file");
+
+	simple_actions_action_group->add_action("new");
+	simple_actions_action_group->add_action("open");
+	simple_actions_action_group->add_action("save");
+	simple_actions_action_group->add_action("save-as");
+	simple_actions_action_group->add_action("save-all");
+	simple_actions_action_group->add_action("export");
+	simple_actions_action_group->add_action("revert");
+	simple_actions_action_group->add_action("import");
+	simple_actions_action_group->add_action("import-sequence");
+	simple_actions_action_group->add_action("render");
+	simple_actions_action_group->add_action("preview");
+	simple_actions_action_group->add_action("close-document");
+	simple_actions_action_group->add_action("quit");
+
+	App::main_window->insert_action_group("menus",simple_menus_action_group);
+	App::main_window->insert_action_group("actions",simple_actions_action_group);
```

Comment:
@rodolforg @ice0 BUT the code is giving segmentation fault ! The moment I comment these two lines and revoke the link of action_groups with main window, code compiles perfectly but the file menu is missing coz I commented the Gtk::Action part !

Reply:
did my best with removing the segmentation fault but couldn't help it ! it would be great if you can rectify my mistake ! and rest of the code is ok according to me ! but we will see if it works or not on seeing the new file menu :)

Reply:
code was not giving seg fault till this commit 

![image](https://user-images.githubusercontent.com/71627983/129497032-f03355de-00e2-4dc5-a25f-44df1a899325.png)

Kindly rectify my mistake, then I will pace up my work :)

Reply:
I didn't test your code yet. But here are some stuff I saw:

**First**, the try-catch you (correctly) surrounded the builder loading: you only wrote one specific "catch": `std::exception`. Documentation says `create_from_string()` can throw `Gtk::BuilderError` and `Glib::MarkupError` : and none is derived from `std::exception`. Add those `catch`.

**Second**, simplify your test: try a menu with a single item first. "Quit" or "New". "Quit" is preferable or you'll have a bit trouble to quit synfig ;) 

**Third**: You create actions three times. Look how docs/examples do:
https://developer-old.gnome.org/gtkmm-tutorial/3.24/sec-buildapp-app-menu.html.en
```c++
  // Add actions and keyboard accelerators for the application menu.
  add_action("preferences", sigc::mem_fun(*this, &ExampleApplication::on_action_preferences));
  add_action("quit", sigc::mem_fun(*this, &ExampleApplication::on_action_quit));
  set_accel_for_action("app.quit", "<Ctrl>Q");

  auto refBuilder = Gtk::Builder::create();
  try
  {
    refBuilder->add_from_resource("/org/gtkmm/exampleapp/app_menu.ui");
  }
  catch (const Glib::Error& ex)
  {
    std::cerr << "ExampleApplication::on_startup(): " << ex.what() << std::endl;
    return;
  }

  auto object = refBuilder->get_object("appmenu");
  auto app_menu = Glib::RefPtr<Gio::MenuModel>::cast_dynamic(object);
  if (app_menu)
```
(On)/After application startup, they create/add (simple) actions to Application (by naming and binding the callback). Only once.
After that, they instance `Builder` and then read menu model from resource (string, in our case).

You didn't bind any callback method. It's okay by now, for initial testing. But once you are able to make a simple menu with a single item working, fix it :)

**Fourth**: you created action groups. Do you ever add these groups to anything? They can be inserted in Gtk::Widget (and Gtk::Window is a Gtk::Widget). But maybe it's better to do stuff without separate action groups first.

**Fifth**: Segmentation fault is, in large majority (or always?), due to null pointer deference.
Search on your newly added code where do you deference a pointer (operator `->`). Are you sure  that object is not null there?
Anyway, you can always run synfig via `gdb` and find where it crashes.

Reply:
Understood ! Will try to make the requested changes as soon as possible :)